src/chain/tracker.rs

```diff
@@ -243,22 +253,39 @@ impl ChainWatcher {
             &chain.asset
         );
         // Remove unwanted assets
-        assets.retain(|name, properties| {
-            tracing::info!(
-                "chain {} has token {} with properties {:?}",
-                &chain.name,
-                &name,
-                &properties
-            );
-            if let Some(native_token) = &chain.native_token {
-                (native_token.name == *name) && (native_token.decimals == specs.decimals)
-            } else {
+        assets = assets
```

Comment:
why replace smart `retain` and `any` with primitives?

Reply:
@fluiderson ^^

Reply:
@Slesarew, this code takes names for assets from the config, not from a chain. The `retain` method doesn't allow the mutating of the key value, hence `filter_map` and then `collect`.

Reply:
and why not from the chain? Taking those from chain ensures they will be accepted by chain and things will match as advertised, does it not?

Reply:
> and why not from the chain? Taking those from chain ensures they will be accepted by chain and things will match as advertised, does it not?
Basically, I just removed checks for asset names from your code. Everything else is the same.

Reply:
If you don't get why, there's no such thing as an asset name for the Assets pallet in terms of the asset transferring, just IDs. Names are just a metadata for humans there. Now, the daemon user can set the same asset names as on the chain or name them on their own (with the checks, this was impossible). The latter feature was very much requested for testing purposes by the frontend team.